BraintreeCore/src/main/java/com/braintreepayments/api/core/BraintreeHttpClient.kt

```diff
@@ -99,48 +62,77 @@ internal class BraintreeHttpClient(
+        if (!validateAuthorization(authorization, callback)) return
+
+        val requestBody = if (authorization is ClientToken) {
             try {
-                JSONObject(data).put(
-                    AUTHORIZATION_FINGERPRINT_KEY,
-                    authorization.authorizationFingerprint
-                ).toString()
+                JSONObject(data)
+                    .put(AUTHORIZATION_FINGERPRINT_KEY, authorization.authorizationFingerprint)
+                    .toString()
             } catch (e: JSONException) {
                 callback?.onResult(null, e)
                 return
             }
         } else {
             data
         }
-        val request = HttpRequest().method("POST").path(path).data(requestData)
-            .addHeader(USER_AGENT_HEADER, "braintree/android/" + BuildConfig.VERSION_NAME)
-        if (isRelativeURL && configuration != null) {
-            request.baseUrl(configuration.clientApiUrl)
+
+        val url = try {
+            assembleUrl(path, configuration)
+        } catch (e: BraintreeException) {
+            callback?.onResult(null, e)
+            return
         }
+
+        val request = OkHttpRequest(
+            method = Method.Post(requestBody),
+            url = url,
+            headers = assembleHeaders(authorization, additionalHeaders)
+        )
+
+        httpClient.sendRequest(request, callback)
+    }
+
+    private fun validateAuthorization(
+        authorization: Authorization?,
+        callback: NetworkResponseCallback?
+    ): Boolean {
+        if (authorization is InvalidAuthorization) {
+            callback?.onResult(null, BraintreeException(authorization.errorMessage))
+            return false
+        }
+        return true
+    }
+
+    @Throws(BraintreeException::class)
+    private fun assembleUrl(path: String, configuration: Configuration?): String {
+        return if (path.startsWith("http")) {
+            path
+        } else {
+            configuration?.clientApiUrl?.plus(path)
+                ?: throw BraintreeException("Braintree HTTP request without configuration cannot have a relative path.")
+        }
+    }
+
+    private fun assembleHeaders(
+        authorization: Authorization?,
+        additionalHeaders: Map<String, String> = emptyMap()
+    ): Map<String, String> {
+        val headers = mutableMapOf(
+            USER_AGENT_HEADER to "braintree/android/" + BuildConfig.VERSION_NAME
+        )
         if (authorization is TokenizationKey) {
-            request.addHeader(CLIENT_KEY_HEADER, authorization.bearer)
+            headers["Client-Key"] = authorization.bearer
         }
-        authorization?.bearer?.let { token -> request.addHeader("Authorization", "Bearer $token") }
-        additionalHeaders.forEach { (name, value) -> request.addHeader(name, value) }
-        httpClient.sendRequest(request, callback)
+        authorization?.bearer?.let { token ->
+            headers["Authorization"] = "Bearer $token"
+        }
```

Comment:
Existing behavior - Does this header need to be added when authorization is TokenizationKey?

Reply:
Just pushed https://github.com/braintree/braintree_android/pull/1407/commits/2007e9a757869871cc3ab14e81be9b033cbce623 updating the auth header based on the Authorization type